### PR TITLE
Do not downcase attributes and use quotes when referring to columns in Postgres.

### LINKED ISSUE #14: Error Creating Index due to Bad Chars
### Error:
```
LINE 1: ...ATE INDEX census_0126 ON census_01 (migration_code-change_in...
                                                             ^
 [SQL: 'CREATE INDEX census_0126 ON census_01 (migration_code-change_in_msa)'] (Background on this error at: http://sqlalche.me/e/f405)
For table: census_01 ERROR while creating index for table census_01 on attributes ['migration_code-change_in_msa']
```
===================
### tmp solution added to my local holoclean/dataset/table:36
bad_chars = re.compile('[^a-zA-Z0-9]+')
renamed_cols = [bad_chars.sub(' ', col).strip().replace(' ', '_') for col in columns]

### PR DESCRIPTION
**Pull request by richardwu**
Closes #14 

Tested this by replacing "HospitalName" with "Hospital-Name" in the hospital dataset and results were reproducible.

### CODE REVIEW

**Review by ScarletGuo — COMMENTED** [review]
May i ask why we do not need downcase attributes anymore?  Will clean data and dirty data be queried in a case-insensitive manner in postgres?

File: dataset/dbengine.py [hunk 1]
```diff
@@ -122,7 +122,8 @@ def create_db_table_from_query(self, name, query):
         return True
 
     def create_db_index(self, name, table, attr_list):
-        stmt = index_template.substitute(idx_title=name, table=table, attr=','.join(attr_list))
+        quoted_attrs = list(map('"{}"'.format, attr_list))
```
Comment (minafarid) [inline dataset/dbengine.py]: Do you think this is more readable?
`['"{}"'.format(attr) for attr in attr_list]`

File: detect/violationdetector.py [hunk 2]
```diff
@@ -80,7 +80,7 @@ def gen_mult_query(self, tbl, c):
         return query
 
     def gen_tid_attr_output(self, res, attr_list):
-        attrs = [x.lower() for x in attr_list]
+        attrs = [x for x in attr_list]
```
Comment (minafarid) [inline detect/violationdetector.py]: doesn't look like this line is needed anymore unless `attr_list` is not already a list

File: evaluate/eval.py [hunk 3]
```diff
@@ -38,8 +38,7 @@ def load_data(self, name, f_path, f_name, get_tid, get_attr, get_val, na_values=
             raw_data['_value_'] = raw_data.apply(get_val, axis=1)
             raw_data = raw_data[['_tid_', '_attribute_', '_value_']]
             # Normalize string to lower-case and strip whitespaces.
-            raw_data['_attribute_'] = raw_data['_attribute_'].apply(lambda x: x.lower())
-            raw_data['_value_'] = raw_data['_value_'].apply(lambda x: x.strip())
+            raw_data['_value_'] = raw_data['_value_'].apply(lambda x: x.lower().strip())
```
Comment (minafarid) [inline evaluate/eval.py]: should we revert this change since it is already being changed in two other PRs?

**Review by minafarid — CHANGES_REQUESTED** [review]

**Comment by minafarid** [comment]
Minor changes requested, good work

File: dataset/dbengine.py (hunk 1 above)
Comment (richardwu) [reply dataset/dbengine.py]: Yes I agree. Changed.

File: detect/violationdetector.py (hunk 2 above)
Comment (richardwu) [reply detect/violationdetector.py]: Yes that's correct, removed.

File: evaluate/eval.py (hunk 3 above)
Comment (richardwu) [reply evaluate/eval.py]: I'll revert this line for now: it should merge fine though as it will display it as a non-change in the other PRs.

**Comment by richardwu** [comment]
@ScarletGuo downcasing is not strictly necessary, but I always like the philosophy that we should not normalize/mutate metadata like attributes' names unless there's a reason to. Rather: we should honour the case/format of the attributes the user supplies in the data.

**Comment by ScarletGuo** [comment]
> @ScarletGuo downcasing is not strictly necessary, but I always like the philosophy that we should not normalize/mutate metadata like attributes' names unless there's a reason to. Rather: we should honour the case/format of the attributes the user supplies in the data.

@richardwu  Yes, I agree esp. when you add quotes to attr names with bad chars. I am asking because there have cases that the clean data has lower-case attributes. just wondering if we make it case-insensitive somewhere when doing comparisons? it's safer to allow for flexibility in the code than changing the dataset. (or, is the customized function provided in test_holoclean.py addressing this case matching issue? )

File: tests/test_holoclean.py [hunk 4]
```diff
@@ -32,9 +32,9 @@ def get_tid(row):
     return row['tid'] - 1
 
 def get_attr(row):
-    return row['attribute'].lower()
+    return row['attribute']
```
Comment (minafarid) [inline tests/test_holoclean.py]: This file conflicts now, but we shouldn't even need to fix it since the new eval API takes the attribute names without the need to lower them

**Review by minafarid — CHANGES_REQUESTED** [review]
I modified the base to `dev` but there are conflicts now. Should be straightforward to fix.

**Comment by richardwu** [comment]
Since we merged #18 which includes the fixes in this PR, this PR no longer does anything. Closing.
